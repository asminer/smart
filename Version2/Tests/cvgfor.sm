
/*
   $Id$
*/

/*
 *   Tests a for statement within a converge statement
 */

	
/*

  x[i] = i
	
*/

# Verbose true
	
# ConvergePrecision 1e-7	

print("Linear system using gauss-seidel, by hand\n");
# UseCurrent true
	
converge {
  print("[");
  for (int i in {1..10}) {
    real x[i] guess i*100;
    real x[i] := cond (i==1, (15-x[10])/5, (6*i-1-x[i-1])/5); 
    print(x[i], cond(i<10, ", ", "")); 
  }
  print("]\n");
}

print("Array x has converged to: [");

for (int i in {1..10}) {
  print(x[i], cond(i<10, ", ", "")); 
}
print("]\n");
	

print("Linear system using Jacobi, by hand\n");
# UseCurrent false
	
converge {
  print("[");
  for (int i in {1..10}) {
    real y[i] guess i*100;
    real y[i] := cond (i==1, (15-y[10])/5, (6*i-1-y[i-1])/5); 
    print(y[i], cond(i<10, ", ", "")); 
  } 
  print("]\n");
}
	
print("Array y has converged to: [");

for (int i in {1..10}) {
  print(y[i], cond(i<10, ", ", "")); 
}
print("]\n");
	
	

