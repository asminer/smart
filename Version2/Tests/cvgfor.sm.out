Linear system using gauss-seidel, by hand
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Array x has converged to: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Linear system using Jacobi, by hand
[100, 200, 300, 400, 500, 600, 700, 800, 900, 1000]
[-197, 200, 300, 400, 500, 600, 700, 800, 900, 1000]
[-197, 41.6, 300, 400, 500, 600, 700, 800, 900, 1000]
[-197, 41.6, -4.92, 400, 500, 600, 700, 800, 900, 1000]
[-197, 41.6, -4.92, 5.584, 500, 600, 700, 800, 900, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 600, 700, 800, 900, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 700, 800, 900, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 800, 900, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 900, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 1000]
[-197, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 41.6, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, -4.92, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 5.584, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 4.6832, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6.06336, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 6.98733, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 7, 8.00253, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 7, 8, 8.99949, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 7, 8, 9, 10.0001]
[0.99998, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Array y has converged to: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
